plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/OpenLineageListenerConfig.java

```diff
@@ -99,10 +103,37 @@ public Optional<String> getNamespace()
     }
 
     @Config("openlineage-event-listener.namespace")
-    @ConfigDescription("Override default namespace for job facet.")
+    @ConfigDescription("Override default namespace for job facet")
     public OpenLineageListenerConfig setNamespace(String namespace)
     {
         this.namespace = Optional.ofNullable(namespace);
         return this;
     }
+
+    public String getJobNameFormat()
+    {
+        return jobNameFormat.orElse(DEFAULT_JOB_NAME_FORMAT);
+    }
+
+    @Config("openlineage-event-listener.job.name-format")
+    @ConfigDescription("Set format for job name facet")
+    public OpenLineageListenerConfig setJobNameFormat(String jobNameFormat)
+    {
+        this.jobNameFormat = Optional.ofNullable(jobNameFormat);
+        if (jobNameFormat == null) {
+            return this;
+        }
+
+        String resolvedJobName = jobNameFormat;
+        for (String substitution : JOB_NAME_SUBSTITUTIONS) {
+            resolvedJobName = resolvedJobName.replace(substitution, "");
+        }
+
+        if (!resolvedJobName.contains("$")) {
+            return this;
+        }
```

Comment:
Can we use a regexp based validator here ?

Reply:
We also use a similar validation here - FormatBasedRemoteQueryModifierConfig

Reply:
I don't wee there this method is called:
https://github.com/trinodb/trino/blob/f373b7461c237583776f8a7ee250dd366b2d61ba/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/logging/FormatBasedRemoteQueryModifierConfig.java#L41-L44
Is this some kind of airlift magic?

Regarding using SessionInterpolatedValues as an example for interpolation:
https://github.com/trinodb/trino/blob/master/lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/logging/SessionInterpolatedValues.java#L31

I see an issue here - some fields are fetched from QueryContext, and others from QueryMetadata, but `InterpolatedValue<T>` generic accepts only one class, and Java doesn't have Union type...

Reply:
This is kind of Airlift magic i.e method with Assert annotation and which starts with `is` would be invoked as a part of validation